pkg/types/aclgroup.go

```diff
@@ -159,6 +161,14 @@ func (d *aclGroupDiffer) createUpdateACLGroup(aclGroup *state.ACLGroup) (*crud.E
 	currentACLGroup, err := d.currentState.ACLGroups.Get(
 		*aclGroup.Consumer.ID, *aclGroup.ID)
 	if errors.Is(err, state.ErrNotFound) {
+		if aclGroup.ID != nil {
+			remoteACLGroup, _ := d.client.ACLs.GetByID(context.TODO(), aclGroup.ID)
```

Comment:
I've missed handling the errors here - 404 is expected, but others are not. 
I'll use `kong.IsNotFoundErr(err)` to fix this.